driver/controller.go

```diff
@@ -288,13 +289,13 @@ func (d *Driver) ControllerUnpublishVolume(ctx context.Context, req *csi.Control
 
 	// TODO:  If node ID is not set, the SP MUST unpublish the volume from all nodes it is published to.
 	log.Info("getting server by hostname")
-	server, err := d.upclouddriver.getServerByHostname(ctx, req.GetNodeId())
+	server, err := d.svc.getServerByHostname(ctx, req.GetNodeId())
```

Comment:
BTW, we have coined an approach in CAPU to find objects using labels. Not to be dependent on what user do with the name of the server in UI. I see that you are checking it with hostname that is ok. But From the reducing API load perspective it is better to query object by labels. The labels is well known and can be sourced from CAPU code-base.

Reply:
I like that approach but I think that with CSI we cannot be sure if (or what) server labels are available, when servers are not managed by UKS.